From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: reintroduce blk_rq_check_limits
Patch-mainline: never, kabi

In 3.12.53, commit 052bd81382379ccaf002ae7a0c3d4ff6f058428b (block:
Always check queue limits for cloned requests), upstream commit
bf4e6b4e757488dee1b6a581f49c7ac34cd217f8 made blk_rq_check_limits
hidden and renamed it to blk_cloned_rq_check_limits.

Reintroduce the function, just as a wrapper to
blk_cloned_rq_check_limits, but don't re-add a declaration into
header as the function should not be called.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 block/blk-core.c |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -1944,6 +1944,15 @@ static int blk_cloned_rq_check_limits(st
 	return 0;
 }
 
+int blk_rq_check_limits(struct request_queue *q, struct request *rq)
+{
+	if (!rq_mergeable(rq))
+		return 0;
+
+	return blk_cloned_rq_check_limits(q, rq);
+}
+EXPORT_SYMBOL_GPL(blk_rq_check_limits);
+
 /**
  * blk_insert_cloned_request - Helper for stacking drivers to submit a request
  * @q:  the queue to submit the request
